lib/fortnox/api/repositories/base.rb

```diff
@@ -88,20 +80,105 @@ def base_url
           base_url
         end
 
+        def client_id
+          client_id = config.client_id
+          raise MissingConfiguration, 'You have to provide your client id.' unless client_id
+          client_id
+        end
+
         def client_secret
           client_secret = config.client_secret
           raise MissingConfiguration, 'You have to provide your client secret.' unless client_secret
-
           client_secret
         end
 
         def config
           Fortnox::API.config
         end
 
-        def token_store_not_found!(store_name)
-          raise MissingConfiguration,
-                "There is no token store named #{store_name}. Available stores are #{config.token_store.keys}."
+        def get_token_store(name)
+          storages = config.storages
+
+          return validate_store(storages[name], name) unless storages.nil?
```

Comment:
```
storage = storages[name] || config.storage
validate_store(storage, name)
```

Reply:
This is a good idea, but you will get `undefined method `[]' for nil:NilClass`